homeassistant/components/cmus/media_player.py

```diff
@@ -80,18 +80,41 @@ def __init__(self, server, password, port, name):
         """Initialize the CMUS device."""
 
         if server:
-            self.cmus = remote.PyCmus(server=server, password=password, port=port)
             auto_name = f"cmus-{server}"
         else:
-            self.cmus = remote.PyCmus()
             auto_name = "cmus-local"
         self._name = name or auto_name
+        self._server = server
+        self._password = password
+        self._port = port
         self.status = {}
+        self.connect()
```

Comment:
We don't want side effects in init methods. Please create and set up the remote device first and then pass that instance to the entity when creating that.

Possibly wrap the device api inside a new class that has a (re)connect method to refresh the connection. It's good to separate the entity api from the device api and not add device specific methods to the entity.

Reply:
Thanks @MartinHjelmare ! It's getting late here - I'll try implementing this on the weekend.

Reply:
@MartinHjelmare - I managed to remove the `connect()` method from `__init()__`.

Rather than wrapping it up inside a new class, I just passed the class from the `remote.PyCmus` package. It should mean a bit less code.

Also, I created a new `reconnect()` function to handle the disconnects within the main device's class.

Was this what you had in mind?